### ISSUE TITLE
[Wanted] Battle testers

### ISSUE THREAD

**Issue by Pyrdacor** [Pyrdacor/Ambermoon.net#59]
First implementation of battles is out there.

Please start testing battles with v1.1.18. Report bugs here so I won't get drowned in oceans of bugs. :D This is still beta and will have many flaws.

Spells are not implemented yet as well as advancing the group. So don't report issues regarding these features.

**Comment by jr-riketz** [comment]
Battle gestartet, funktioniert ganz gut Bewegungen soweit ok.
nach ca. 1,5 Minuten stürzt das Spiel unverhofft ab.
nach nun 10 Kämpfen mal nach einem Lauf Befehl, mal nach einem Angriffsbefehl. willkürlich

**Comment by Pyrdacor** [comment]
Thanks for reporting this. Is there an error message after the crash?

**Comment by jr-riketz** [comment]
> Thanks for reporting this. Is there an error message after the crash?

Its a Crash no Error Message

**Comment by Pyrdacor** [comment]
1.1.19 fixes some things. Please give it a try.

**Comment by jr-riketz** [comment]
![stuck_after_escape](https://user-images.githubusercontent.com/7780378/99585063-1cf57f80-29e6-11eb-9681-3bc488cfc05b.png)

Battle Crash is gone! nice Job

New Problem:
The enemy has run away.
Subsequently, the fight was not ended.
When I tried to run away with all the characters, I got stuck in this window.
I couldn't close it so I had to quit the game.

Greetz J

**Comment by jr-riketz** [comment]
> ![stuck_after_escape](https://user-images.githubusercontent.com/7780378/99585063-1cf57f80-29e6-11eb-9681-3bc488cfc05b.png)
> 
> Battle Crash is gone! nice Job
> 
> New Problem:
> The enemy has run away.
> Subsequently, the fight was not ended.
> When I tried to run away with all the characters, I got stuck in this window.
> I couldn't close it so I had to quit the game.
> 
> Greetz J


i tried about 20 fights, this scenario where the enemy runs away is apparently the trigger.

The fight mode should actually be ended here.

**Comment by Pyrdacor** [comment]
Alright. I will look into it tomorrow. Thanks for testing.

**Comment by Pyrdacor** [comment]
I fixed your issue and others regarding fleeing as well. But next release will only come in a few days or maybe tomorrow.

**Comment by Pyrdacor** [comment]
The battle should end now when all monsters or the party flee.

I added the first test spell. You can click on the "use magic" button with any party member. Then click on a target monster on the battle field. The spell fire ball is then performed.

Note: There is no spell list popup yet. And the spell does no real damage at the moment. But the animations should be displayed.

**Comment by jr-riketz** [comment]
Nice Spell animation, thanks for that.

Issue i just saw:

I killed a bandit in town and after he died his disappearing animation was two blocks behind his original position.

I tried to capture this in a picture, I think you know what I mean.
PS: The second picture was from another fight in which this also happened, hence the shift to the right, the disappearing animation.
![before](https://user-images.githubusercontent.com/7780378/99882338-f589f700-2c1f-11eb-9010-6b73d76a0a0a.jpg)
![after](https://user-images.githubusercontent.com/7780378/99882339-f753ba80-2c1f-11eb-8314-c41cc504fe2c.JPG)


Problem is the Same in the last 20 Battles.

Greetz J

**Comment by jr-riketz** [comment]
Another little question:
At the beginning of the fireball magic, the play area is briefly colored red.
I wanted it that way, I can't even remember it.
;-)

**Comment by jr-riketz** [comment]
> The battle should end now when all monsters or the party flee.
> 
> I added the first test spell. You can click on the "use magic" button with any party member. Then click on a target monster on the battle field. The spell fire ball is then performed.
> 
> Note: There is no spell list popup yet. And the spell does no real damage at the moment. But the animations should be displayed.


Nice that you thought that warriors do not cast spells and that it is deactivated for these players.

**Comment by Pyrdacor** [comment]
I will have a look at the displacement issue in the coming week
Thanks for reporting.

Yeah the screen is colored red briefly in original game. So I added it as well. The screen is briefly colored blue for ice spells. This is also prepared already.

About the warrior cast thing. Yeah I try my best to implement every single detail of the original game. And there are lots of them. Most things are drawn pixel perfect (at the same pixel as in original) and I do this by taking screenshots and measure it by hand. So it is much more work than it may look like. I really love the game so I want the best for the rework as well and give all the great memories to all who cherished the original or love retro games and their details.
I am glad that you pointed it out and like it. This is the reason why I am doing this. :)

**Comment by Pyrdacor** [comment]
I could need some help in finding out the exact damage values of spells. I did already for the 4 fire spells but the rest needs testing. I also don't know if the damage is dependent on the caster and/or the target. The latter one didn't seem the case for my fire spell tests which I performed on gargoyles, gigantula, poison spider and small spider.

**Comment by Thallyrion** [comment]
What exactly did you find out about the damage values of the Fire spells? Do they correspond to the damage values stated in the manual?

I just tested a battle and got a crash:

Object reference not set to an instance of an object.
   at Ambermoon.Game.<>c__DisplayClass261_0.<ShowBattleWindow>g__EndBattle|8(BattleEndInfo battleEndInfo)
   at Ambermoon.Game.<>c__DisplayClass261_1.<ShowBattleWindow>b__9()
   at Ambermoon.Game.ShowBattleLoot(BattleEndInfo battleEndInfo, Action closeAction)
   at Ambermoon.Game.<>c__DisplayClass261_0.<ShowBattleWindow>b__4(BattleEndInfo battleEndInfo)
   at Ambermoon.Battle.KillMonster(PartyMember attacker, Character target)
   at Ambermoon.Battle.<>c__DisplayClass91_0.<HandleCharacterDeath>b__1()
   at Ambermoon.Battle.<>c__DisplayClass116_0.<PlayBattleEffectAnimation>g__FinishEffect|0()
   at Ambermoon.Battle.<>c__DisplayClass117_0.<PlayBattleEffectAnimation>g__EffectAnimationFinished|0()
   at Ambermoon.Render.BattleAnimation.Update(UInt32 ticks)
   at Ambermoon.Battle.Update(UInt32 battleTicks)
   at Ambermoon.Game.UpdateBattle()
   at Ambermoon.Game.Update(Double deltaTime)
   at Silk.NET.Windowing.Desktop.GlfwWindow.RaiseUpdateFrame()
   at Silk.NET.Windowing.Desktop.GlfwWindow.DoUpdate()
   at Silk.NET.Windowing.Common.WindowExtensions.<>c__DisplayClass2_0.<Run>b__0()
   at Silk.NET.Windowing.Desktop.GlfwWindow.Run(Action onFrame)
   at Silk.NET.Windowing.Common.WindowExtensions.Run(IView view)
   at Ambermoon.GameWindow.Run(Configuration configuration)
   at Ambermoon.Program.Main()

**Comment by Pyrdacor** [comment]
@Thallyrion: About the crash: How many members had your party? Less than 6?

**Comment by Pyrdacor** [comment]
I couldn't find any information about spell damage in the manual nor on the web.

**Comment by Thallyrion** [comment]
Indeed i think i mixed it up with another game where spelldamage was included in the manual.

My Party had 2 Members

**Comment by Pyrdacor** [comment]
The spell damage in Amberstar seems to be very different. In Ambermoon the strongest spells are the ice spells, then the fire spells, then wind and then earth. In Amberstar fire and ice/water spells are weak in relation to wind or earth spells. Fireball is stated as 4-8 damage. In Ambermoon I found a range of 40-80. Wind devil is given with 16-32 in Amberstar but I found a range of 10-15 in Ambermoon. So I guess we could not rely on Ambermoon here.
